### ISSUE TITLE
Falco 0.36.1 and 0.36.2 segfault on startup when gvisor configuration is specified

### ISSUE THREAD

**Issue by thundergolfer** [falcosecurity/falco#2896]
**Describe the bug**

Cannot startup `falco`, segmentation fault occurs after logging 

```
Wed Nov  1 20:23:33 2023: Opening 'syscall' source with gVisor. Configuration path: /etc/falco/pod-init.json
Segmentation fault
```

**How to reproduce it**

```
sudo  falco -c /etc/falco/falco.yaml   --gvisor-config /etc/falco/pod-init.json   --gvisor-root /run/user/1000/runsc
```

I don't think the value of `--gvisor-root` matters. Can also get the fault with: 

```
sudo falco  -c /etc/falco/falco.yaml   --gvisor-config /etc/falco/pod-init.json   --gvisor-root /var/run/docker/runtime-runc/moby
```

The `/etc/falco/pod-init.json` file was generated with `falco --gvisor-generate-config`.

`/etc/falco/falco.yaml` is produced with

```
sudo bash -c "cat << EOF > $FALCO_CONFIG_PATH
rules_file:
  - /etc/falco/falco_rules.yaml
  - /etc/falco/falco_rules.local.yaml
  - /etc/falco/rules.d

watch_config_files: true
time_format_iso_8601: false

priority: debug

json_output: true
json_include_output_property: false
json_include_tags_property: false
buffered_outputs: false

rule_matching: first

# Redirect logs to standard output.
stdout_output:
  enabled: true

syslog_output:
  enabled: true

# Useful when running Falco in Kubernetes.
webserver:
  enabled: false

# Falco program logging.
log_stderr: true
log_syslog: true
log_level: info
libs_logger:
  enabled: false
  severity: debug

# Metrics reporting.
metrics:
  enabled: false

grpc_output:
  enabled: false

EOF
"
```

**Expected behaviour**

**Screenshots**

<kbd>
<img width="1350" alt="image" src="https://github.com/falcosecurity/falco/assets/12058921/167aa449-395f-48b5-bb78-49e17f459a4f">
</kbd>

**Environment**

- Falco version:

```
falco --version
Wed Nov  1 20:30:20 2023: Falco version: 0.36.1 (x86_64)
Wed Nov  1 20:30:20 2023: Falco initialized with configuration file: /etc/falco/falco.yaml
Falco version: 0.36.1
Libs version:  0.13.2
Plugin API:    3.1.0
Engine:        26
Driver:
  API version:    5.0.0
  Schema version: 2.0.0
  Default driver: 6.0.1+driver
```

- System info:

```json
{
  "machine": "x86_64",
  "nodename": "ip-10-1-8-45",
  "release": "5.15.0-1044-aws",
  "sysname": "Linux",
  "version": "#49~20.04.1-Ubuntu SMP Mon Aug 21 17:09:32 UTC 2023"
}
```

- Cloud provider or hardware configuration:
- OS:

```
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

- Kernel: `Linux ip-10-1-8-45 5.15.0-1044-aws #49~20.04.1-Ubuntu SMP Mon Aug 21 17:09:32 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux`

- Installation method: From https://download.falco.org tarballs

**Additional context**

**Comment by FedeDP** [comment]
Thanks for opening this issue; i tagged people that worked on the gvisor scap engine ;) hopefully they can quickly discover the issue and fix it!

**Comment by LucaGuerra** [comment]
I can confirm that the upcoming Falco 0.37.0 version contains several gVisor updates (newer runsc version support, new supported event types...) and fixes this issue. ETA is end of January 2024.

**Comment by Andreagit97** [comment]
This should be solved by [Falco 0.37.0](https://github.com/falcosecurity/falco/releases/tag/0.37.0)! Feel free to reopen if this is still an issue
